opentelemetry-etw-logs/src/processor.rs

```diff
@@ -118,21 +153,32 @@ impl ProcessorBuilder {
     }
 
     fn validate(&self) -> Result<(), Box<dyn Error>> {
-        validate_provider_name(self.options.provider_name())?;
+        validate_provider_name(
+            self.options.provider_name(),
+            self.cross_compat_provider_name,
+        )?;
         Ok(())
     }
 }
 
-fn validate_provider_name(provider_name: &str) -> Result<(), Box<dyn Error>> {
+fn validate_provider_name(
+    provider_name: &str,
+    cross_compat_provider_name: bool,
+) -> Result<(), Box<dyn Error>> {
     if provider_name.is_empty() {
         return Err("Provider name must not be empty.".into());
     }
     if provider_name.len() >= 234 {
         return Err("Provider name must be less than 234 characters long.".into());
     }
+
+    if cross_compat_provider_name && provider_name.contains('-') {
+        return Err("Provider name must not contain hyphens ('-') when cross-compatibility with UserEvents is enabled. Use ProviderBuilder::builder_etw_compat_only() to enable ETW only compatibility instead.".into());
+    }
+
     if !provider_name
         .chars()
-        .all(|c| c.is_ascii_alphanumeric() || c == '_')
+        .all(|c| c.is_ascii_alphanumeric() || c == '_' || (!cross_compat_provider_name && c == '-'))
     {
         return Err("Provider name must contain only ASCII alphanumeric characters or '_'.".into());
```

Comment:
This message is invalid in the case of enabled "-" support, and why do we have a separate message for the same thing above?